### ISSUE TITLE
Installation commands reference /v2 of protoc-gen-grpc-gateway and protoc-gen-openapiv2, they don't exist

### ISSUE THREAD

**Issue by javorszky** [grpc-ecosystem/grpc-gateway#1992]
## 📚 Documentation

Following the [documentation on installation](https://github.com/grpc-ecosystem/grpc-gateway#installation), and running the following two commands to install the tools globally on my computer results in the following outputs:

```
$ go get -u github.com/grpc-ecosystem/grpc-gateway/v2/protoc-gen-grpc-gateway
cannot find package "github.com/grpc-ecosystem/grpc-gateway/v2/protoc-gen-grpc-gateway" in any of:
	/usr/local/Cellar/go/1.15.8/libexec/src/github.com/grpc-ecosystem/grpc-gateway/v2/protoc-gen-grpc-gateway (from $GOROOT)
	/Users/<myuser>/go/src/github.com/grpc-ecosystem/grpc-gateway/v2/protoc-gen-grpc-gateway (from $GOPATH)
```

```
$ go get -u github.com/grpc-ecosystem/grpc-gateway/v2/protoc-gen-openapiv2
cannot find package "github.com/grpc-ecosystem/grpc-gateway/v2/protoc-gen-openapiv2" in any of:
	/usr/local/Cellar/go/1.15.8/libexec/src/github.com/grpc-ecosystem/grpc-gateway/v2/protoc-gen-openapiv2 (from $GOROOT)
	/Users/<myuser>/go/src/github.com/grpc-ecosystem/grpc-gateway/v2/protoc-gen-openapiv2 (from $GOPATH)
```

The documentation references a v2 directory which doesn't exist in the [current master](https://github.com/grpc-ecosystem/grpc-gateway/tree/1ae6b3cdb14b220773ada65b9ce392644feedf07) branch of the repository.

Dropping the `/v2` parts from the paths leads to a successful `go get` for both of them.

**Comment by javorszky** [comment]
turns out go wasn't in GO111MODULE=on mode

that solved this issue

**Comment by javorszky** [comment]
The above happens when the `GO111MODULE` was not set, or was set to `off`.

If the `GO111MODULE=on` is specified, the go gets for the /v2 paths succeed, however the `go installs` fail with this:

```
cannot find module providing package github.com/grpc-ecosystem/grpc-gateway/v2/protoc-gen-grpc-gateway: working directory is not part of a module
cannot find module providing package github.com/grpc-ecosystem/grpc-gateway/v2/protoc-gen-openapiv2: working directory is not part of a module
```

I can work around this, but I still feel documentation could be improved.

**Comment by johanbrandhorst** [comment]
Hi Gabor, sorry for the confusion, how would you like us to update the docs exactly? Would you be willing to make a docs contribution to help others?

**Comment by javorszky** [comment]
I'll write up a PR with documentation change to make this clearer later today. 🙂

**Comment by adambabik** [comment]
I tried to reproduce it and setting `GO111MODULE=on` is in fact not enough. The documentation says:

> The following instructions assume you are using Go Modules for dependency management.

which means there is `go.mod` file in the working directory. Only then it works as expected but honestly that was my understanding reading this paragraph. Also, this is true for Go 1.15. The recently released Go 1.16 will change this and `go.mod` won't be required. In theory, this special file with `+build tools` constraint won't be needed as well, however, I think it still will be useful for keeping the versions in sync. More about [modules in Go 1.16](https://blog.golang.org/go116-module-changes).

**Comment by javorszky** [comment]
Apologies, yesterday I ended up not having enough time to create a PR. It's now ready and open in #1997. Let me know if you'd like me to reword anything on that.
